cosmos/runtime/txpool/ante.go

```diff
@@ -60,27 +62,24 @@ func (m *Mempool) shouldEjectFromCometMempool(
 	if tx == nil {
 		return false
 	}
-	txStatus := m.txStatus(tx.Hash())
+	txHash := tx.Hash()
 
 	// Ejection conditions
 	// 1. If the transaction has been included in a block.
-	// 2. If the transaction is unknown to the node.
-	// 3. If the transaction has been in the mempool for longer than the configured timeout.
-	return txStatus == txpool.TxStatusIncluded || txStatus == txpool.TxStatusUnknown ||
-		currentTime.Sub(tx.Time()) > m.lifetime
+	// 2. If the transaction has been in the mempool for longer
+	// than the configured timeout. (txs view).
+	// 3, If the transaction has been in the mempool for longer
+	// than the configured timeout. (our view).
+	m.receivedFromCometAtMu.RLock()
+	cometTime := m.receivedFromCometAt[txHash]
+	m.receivedFromCometAtMu.RUnlock()
+	return m.inCanonicalChain(txHash) ||
+		currentTime.Sub(tx.Time()) > m.lifetime ||
```

Comment:
just double checked `tx.Time()` is set on line 47 in `wet.Unwrap()` (because a new Transaction object gets created and `time` field is set in `UnmarshalBinary`). `currentTime` is `ctx.BlockTime()`, which will actually be earlier so this case will always be false